File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasicExtensions.cs
using System;
using Microsoft.CodeAnalysis.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace Microsoft.CodeAnalysis
{
	[StandardModule]
	public sealed class VisualBasicExtensions
	{
		public static bool IsKind(this SyntaxTrivia trivia, SyntaxKind kind)
		{
			return trivia.RawKind == (int)kind;
		}

		public static bool IsKind(this SyntaxToken token, SyntaxKind kind)
		{
			return token.RawKind == (int)kind;
		}

		public static bool IsKind(this SyntaxNode node, SyntaxKind kind)
		{
			if (node != null)
			{
				return node.RawKind == (int)kind;
			}
			return false;
		}

		public static bool IsKind(this SyntaxNodeOrToken nodeOrToken, SyntaxKind kind)
		{
			return nodeOrToken.RawKind == (int)kind;
		}

		public static int IndexOf<TNode>(this SyntaxList<TNode> list, SyntaxKind kind) where TNode : SyntaxNode
		{
			return list.IndexOf((int)kind);
		}

		public static bool Any<TNode>(this SyntaxList<TNode> list, SyntaxKind kind) where TNode : SyntaxNode
		{
			return list.IndexOf((int)kind) >= 0;
		}

		public static int IndexOf<TNode>(this SeparatedSyntaxList<TNode> list, SyntaxKind kind) where TNode : SyntaxNode
		{
			return list.IndexOf((int)kind);
		}

		public static bool Any<TNode>(this SeparatedSyntaxList<TNode> list, SyntaxKind kind) where TNode : SyntaxNode
		{
			return list.IndexOf((int)kind) >= 0;
		}

		public static int IndexOf(this SyntaxTriviaList list, SyntaxKind kind)
		{
			return list.IndexOf((int)kind);
		}

		public static bool Any(this SyntaxTriviaList list, SyntaxKind kind)
		{
			return list.IndexOf((int)kind) >= 0;
		}

		public static int IndexOf(this SyntaxTokenList list, SyntaxKind kind)
		{
			return list.IndexOf((int)kind);
		}

		public static bool Any(this SyntaxTokenList list, SyntaxKind kind)
		{
			return list.IndexOf((int)kind) >= 0;
		}

		internal static SyntaxToken FirstOrDefault(this SyntaxTokenList list, SyntaxKind kind)
		{
			int num = list.IndexOf((int)kind);
			if (num < 0)
			{
				return default(SyntaxToken);
			}
			return list[num];
		}

		internal static SyntaxToken First(this SyntaxTokenList list, SyntaxKind kind)
		{
			int num = list.IndexOf((int)kind);
			if (num < 0)
			{
				throw new InvalidOperationException();
			}
			return list[num];
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasicExtensions.vb
Imports Microsoft.CodeAnalysis.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Runtime.CompilerServices

Namespace Microsoft.CodeAnalysis
	Public Module VisualBasicExtensions
		<Extension>
		Public Function Any(Of TNode As SyntaxNode)(ByVal list As SyntaxList(Of TNode), ByVal kind As SyntaxKind) As Boolean
			Return list.IndexOf(CInt(kind)) >= 0
		End Function

		<Extension>
		Public Function Any(Of TNode As SyntaxNode)(ByVal list As SeparatedSyntaxList(Of TNode), ByVal kind As SyntaxKind) As Boolean
			Return list.IndexOf(CInt(kind)) >= 0
		End Function

		<Extension>
		Public Function Any(ByVal list As SyntaxTriviaList, ByVal kind As SyntaxKind) As Boolean
			Return list.IndexOf(CInt(kind)) >= 0
		End Function

		<Extension>
		Public Function Any(ByVal list As SyntaxTokenList, ByVal kind As SyntaxKind) As Boolean
			Return list.IndexOf(CInt(kind)) >= 0
		End Function

		<Extension>
		Friend Function First(ByVal list As SyntaxTokenList, ByVal kind As SyntaxKind) As SyntaxToken
			Dim num As Integer = list.IndexOf(CInt(kind))
			If (num < 0) Then
				Throw New InvalidOperationException()
			End If
			Return list(num)
		End Function

		<Extension>
		Friend Function FirstOrDefault(ByVal list As SyntaxTokenList, ByVal kind As SyntaxKind) As SyntaxToken
			Dim num As Integer = list.IndexOf(CInt(kind))
			If (num < 0) Then
				Return New SyntaxToken()
			End If
			Return list(num)
		End Function

		<Extension>
		Public Function IndexOf(Of TNode As SyntaxNode)(ByVal list As SyntaxList(Of TNode), ByVal kind As SyntaxKind) As Integer
			Return list.IndexOf(CInt(kind))
		End Function

		<Extension>
		Public Function IndexOf(Of TNode As SyntaxNode)(ByVal list As SeparatedSyntaxList(Of TNode), ByVal kind As SyntaxKind) As Integer
			Return list.IndexOf(CInt(kind))
		End Function

		<Extension>
		Public Function IndexOf(ByVal list As SyntaxTriviaList, ByVal kind As SyntaxKind) As Integer
			Return list.IndexOf(CInt(kind))
		End Function

		<Extension>
		Public Function IndexOf(ByVal list As SyntaxTokenList, ByVal kind As SyntaxKind) As Integer
			Return list.IndexOf(CInt(kind))
		End Function

		<Extension>
		Public Function IsKind(ByVal trivia As SyntaxTrivia, ByVal kind As SyntaxKind) As Boolean
			Return trivia.RawKind = CInt(kind)
		End Function

		<Extension>
		Public Function IsKind(ByVal token As SyntaxToken, ByVal kind As SyntaxKind) As Boolean
			Return token.RawKind = CInt(kind)
		End Function

		<Extension>
		Public Function IsKind(ByVal node As SyntaxNode, ByVal kind As SyntaxKind) As Boolean
			If (node Is Nothing) Then
				Return False
			End If
			Return node.RawKind = CInt(kind)
		End Function

		<Extension>
		Public Function IsKind(ByVal nodeOrToken As SyntaxNodeOrToken, ByVal kind As SyntaxKind) As Boolean
			Return nodeOrToken.RawKind = CInt(kind)
		End Function
	End Module
End Namespace